jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -49,41 +85,273 @@ def marks(self):
+                self.reset_coords_display()
+
+        elif isinstance(viewer, MosvizProfile2DView):
+            coords_status = False
+
+        if coords_status:
+            celestial_coordinates = sky.to_string('hmsdms', precision=4, pad=True).split()
+            celestial_coordinates_deg = sky.to_string('decimal', precision=10, pad=True).split()
+            world_ra = celestial_coordinates[0]
+            world_dec = celestial_coordinates[1]
+            world_ra_deg = celestial_coordinates_deg[0]
+            world_dec_deg = celestial_coordinates_deg[1]
+
+            if "nan" in (world_ra, world_dec, world_ra_deg, world_dec_deg):
+                self.reset_coords_display()
+
+            self.row2_title = 'World'
+            self.row2_text = f'{world_ra} {world_dec} (ICRS)'
+            self.row2_unreliable = unreliable_world
+            self.row3_title = ''
+            self.row3_text = f'{world_ra_deg} {world_dec_deg} (deg)'
+            self.row3_unreliable = unreliable_world
+
+        maxsize = int(np.ceil(np.log10(np.max(image.shape)))) + 3
+        fmt = 'x={0:0' + str(maxsize) + '.1f} y={1:0' + str(maxsize) + '.1f}'
+        self.row1a_title = 'Pixel'
+        self.row1a_text = (fmt.format(x, y))
+        self.row1_unreliable = unreliable_pixel
+
+        # Extract data values at this position.
+        # TODO: for now we just use the first visible layer but we should think
+        # of how to display values when multiple datasets are present.
+
+        # Extract data values at this position.
+        # Check if shape is [x, y, z] or [y, x] and show value accordingly.
+        if image.ndim == 3:
+            # needed for cubeviz
+            ix_shape = 0
+            iy_shape = 1
+        elif image.ndim == 2:
+            ix_shape = 1
+            iy_shape = 0
+        else:  # pragma: no cover
+            raise ValueError(f'does not support ndim={image.ndim}')
+
+        if (-0.5 < x < image.shape[ix_shape] - 0.5 and -0.5 < y < image.shape[iy_shape] - 0.5
+                and hasattr(active_layer, 'attribute')):
+            attribute = active_layer.attribute
+            if isinstance(viewer, (ImvizImageView, MosvizImageView, MosvizProfile2DView)):
+                value = image.get_data(attribute)[int(round(y)), int(round(x))]
+                unit = image.get_component(attribute).units
+            elif isinstance(viewer, CubevizImageView):
+                arr = image.get_component(attribute).data
+                unit = image.get_component(attribute).units
+                if image.ndim == 3:
+                    value = arr[int(round(x)), int(round(y)), viewer.state.slices[-1]]
+                else:  # 2
+                    value = arr[int(round(y)), int(round(x))]
+            self.row1b_title = 'Value'
+            self.row1b_text = f'{value:+10.5e} {unit}'
+        else:
+            self.row1b_title = ''
+            self.row1b_text = ''
+
+    def _spectrum_viewer_update(self, viewer, x, y):
```

Comment:
Thinking ahead to Timeviz, do you think the light curve viewer will be similar enough that we can call this `_profile_viewer_update` and use it for that as well?

Reply:
definitely something to keep in mind... we might be able to at least share a good chunk of the code between the two with maybe just a few if-statements for handling the differently-named attributes